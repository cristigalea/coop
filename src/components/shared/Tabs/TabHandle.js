import React from 'react';
import PropTypes from 'prop-types';

import { StyleSheet, css } from 'aphrodite';

const TabHandle = ({isActive = false, onClick, children}) => {
    const className = isActive ? css(styles.handle, styles.handleActive) : css(styles.handle);
    return <li onClick={onClick} className={className}>{children}</li>;
}

TabHandle.propTypes = {
    isActive: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    children: PropTypes.any.isRequired
}

export default TabHandle;

const baseColor = '#054f9b';
const accentColor = '#fff';

const styles = StyleSheet.create({
    handle: {
        flex: 1,
        fontSize: '15px',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        textAlign: 'center',
        fontWeight: 'bold',
        cursor: 'pointer',
        color: baseColor,
        borderLeft: `2px solid ${baseColor}`,
        ':first-of-type': {
            borderLeft: 0
        },
        '@media screen and (max-width: 640px)': {
            fontSize: '12px'
        },
        '@media screen and (max-width: 500px)': {
            fontSize: '10px'
        }
    },
    handleActive: {
        backgroundColor: baseColor,
        color: accentColor
    }
});