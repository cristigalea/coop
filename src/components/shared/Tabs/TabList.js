import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, css } from 'aphrodite';

import TabHandle from './TabHandle';

const TabList = ({tabs, selectedTabIndex = 0, handleTabChange}) => {
    const handles = tabs.map((tab, index) => {
        
        return (
            <TabHandle 
                key={`tab-${index}`} 
                onClick={() => handleTabChange(index)} 
                isActive={index === selectedTabIndex}
            >
                {tab}
            </TabHandle>
        );
    })

    return (
        <ul className={css(styles.list)}>
            {handles}
        </ul>
    );
};

TabList.propTypes = {
    tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
    selectedTabIndex: PropTypes.number,
    handleTabChange: PropTypes.func.isRequired
}

export default TabList;

const baseColor = '#054f9b';
const tabHeight = 70;

const styles = StyleSheet.create({
    list: {
        listStyle: 'none',
        paddingLeft: 0,
        display: 'flex',
        minHeight: `${tabHeight}px`,
        border: `2px solid ${baseColor}`,
        borderRadius: `${tabHeight/2}px`,
        overflow: 'hidden'
    }
});