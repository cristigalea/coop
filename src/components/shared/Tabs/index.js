import React, { Component } from 'react';
import { StyleSheet, css } from 'aphrodite';
import PropTypes from 'prop-types';

import TabList from './TabList';

export default class Tabs extends Component { 

    static propTypes = {
        tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
        children: PropTypes.any.isRequired
    };

    state = {
        selectedTabIndex: 0
    };

    handleTabChange = index => {
        this.setState({
            selectedTabIndex: index
        });
    }

    render() {
        const { tabs, children } = this.props;
        const { selectedTabIndex } = this.state;

        // in case not enough children are passed
        const content = children[selectedTabIndex] || <div>No content available</div>; 

        return (
            <div>
                <nav>
                    <TabList 
                        tabs={tabs} 
                        selectedTabIndex={selectedTabIndex}
                        handleTabChange={this.handleTabChange}
                    />
                </nav>
                <div className={css(styles.content)}>
                    {content}
                </div>
            </div>
        )
    }
}

const styles = StyleSheet.create({
    content: {
        padding: '20px'
    }
});