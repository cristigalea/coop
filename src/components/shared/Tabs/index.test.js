import React from 'react';
import {shallow} from 'enzyme';

import Tabs from './';
import { StyleSheetTestUtils} from 'aphrodite';

describe('<Tabs />', () => {
    beforeEach(() => {
        StyleSheetTestUtils.suppressStyleInjection();
    });
      
    afterEach(() => {
        StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    });

    it('should match the snapshot', () => {
        const component = shallow(<Tabs tabs={['Tab1', 'Tab2']}><div>Tab1</div><div>Tab2</div></Tabs>);
        expect(component).toMatchSnapshot();
    });
});