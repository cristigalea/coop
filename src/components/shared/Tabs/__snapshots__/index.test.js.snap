// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tabs
    tabs={
      Array [
        "Tab1",
        "Tab2",
      ]
    }
  >
    <div>
      Tab1
    </div>
    <div>
      Tab2
    </div>
  </Tabs>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav>
          <TabList
            handleTabChange={[Function]}
            selectedTabIndex={0}
            tabs={
              Array [
                "Tab1",
                "Tab2",
              ]
            }
          />
        </nav>,
        <div
          className="content_1rece6a"
        >
          <div>
            Tab1
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TabList
            handleTabChange={[Function]}
            selectedTabIndex={0}
            tabs={
              Array [
                "Tab1",
                "Tab2",
              ]
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleTabChange": [Function],
            "selectedTabIndex": 0,
            "tabs": Array [
              "Tab1",
              "Tab2",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            Tab1
          </div>,
          "className": "content_1rece6a",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tab1",
          },
          "ref": null,
          "rendered": "Tab1",
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav>
            <TabList
              handleTabChange={[Function]}
              selectedTabIndex={0}
              tabs={
                Array [
                  "Tab1",
                  "Tab2",
                ]
              }
            />
          </nav>,
          <div
            className="content_1rece6a"
          >
            <div>
              Tab1
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <TabList
              handleTabChange={[Function]}
              selectedTabIndex={0}
              tabs={
                Array [
                  "Tab1",
                  "Tab2",
                ]
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleTabChange": [Function],
              "selectedTabIndex": 0,
              "tabs": Array [
                "Tab1",
                "Tab2",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              Tab1
            </div>,
            "className": "content_1rece6a",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tab1",
            },
            "ref": null,
            "rendered": "Tab1",
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
